### ISSUE TITLE
Python.exe Spawning, High/Maxing CPU, Traceback Master-Side, Tornado Related?

### ISSUE THREAD

**Issue by TheVakman** [saltstack/salt#48241]
### Description of Issue/Question
If the Salt Master is running, the Salt Minions max out the CPU by spawning Python.exe processes over and over. Additionally, a traceback is thrown for Windows 10 (maybe other) minions on the Salt Master side log.

This appears to be since I moved to a new host with 2018.3.1 Py3 however, there was a time it was certainly functional.

If I stop the Salt Master, the processes stop spawning and my systems return to functional CPU levels.

I've only noticed this problem on Windows 10 v1607 devices thus far. Our Linux clients seem to be unaffected.

### Setup
I'm not sure what should be provided here, any states that are attempted to apply cause this problem.

### Steps to Reproduce Issue
Start Master, Start Minion.

2018-06-21 11:56:13,769 [salt.master      :1515][ERROR   ][10481] Received minion error from [MINION]: The minion function caused an exception
```
2018-06-20 14:56:11,609 [tornado.general  :417 ][ERROR   ][8846] Uncaught exception, closing connection.
Traceback (most recent call last):
  File "/usr/lib64/python3.4/site-packages/zmq/eventloop/zmqstream.py", line 414, in _run_callback
    callback(*args, **kwargs)
  File "/usr/lib64/python3.4/site-packages/tornado/stack_context.py", line 274, in null_wrapper
    return fn(*args, **kwargs)
  File "/usr/lib64/python3.4/site-packages/zmq/eventloop/zmqstream.py", line 191, in <lambda>
    self.on_recv(lambda msg: callback(self, msg), copy=copy)
  File "/usr/lib64/python3.4/site-packages/tornado/gen.py", line 285, in wrapper
    yielded = next(result)
  File "/usr/lib/python3.4/site-packages/salt/transport/zeromq.py", line 687, in handle_message
    ret, req_opts = yield self.payload_handler(payload)
  File "/usr/lib64/python3.4/site-packages/tornado/gen.py", line 270, in wrapper
    result = func(*args, **kwargs)
  File "/usr/lib/python3.4/site-packages/salt/master.py", line 1010, in _handle_payload
    'clear': self._handle_clear}[key](load)
  File "/usr/lib/python3.4/site-packages/salt/master.py", line 1064, in _handle_aes
    ret = self.aes_funcs.run_func(data['cmd'], data)
  File "/usr/lib/python3.4/site-packages/salt/master.py", line 1776, in run_func
    ret = getattr(self, func)(load)
  File "/usr/lib/python3.4/site-packages/salt/master.py", line 1481, in _pillar
    data = pillar.compile_pillar()
  File "/usr/lib/python3.4/site-packages/salt/pillar/__init__.py", line 1003, in compile_pillar
    pillar, errors = self.render_pillar(matches)
  File "/usr/lib/python3.4/site-packages/salt/pillar/__init__.py", line 843, in render_pillar
    pstate, mods, err = self.render_pstate(sls, saltenv, mods)
  File "/usr/lib/python3.4/site-packages/salt/pillar/__init__.py", line 736, in render_pstate
    **defaults)
  File "/usr/lib/python3.4/site-packages/salt/template.py", line 93, in compile_template
    ret = render(input_data, saltenv, sls, **render_kwargs)
  File "/usr/lib/python3.4/site-packages/salt/renderers/gpg.py", line 338, in render
    log.debug('Reading GPG keys from: %s', _get_key_dir())
  File "/usr/lib/python3.4/site-packages/salt/renderers/gpg.py", line 248, in _get_key_dir
    if 'config.get' in __salt__:
  File "/usr/lib64/python3.4/_collections_abc.py", line 428, in __contains__
    self[key]
  File "/usr/lib/python3.4/site-packages/salt/loader.py", line 1143, in __getitem__
    func = super(LazyLoader, self).__getitem__(item)
  File "/usr/lib/python3.4/site-packages/salt/utils/lazy.py", line 99, in __getitem__
    if self._load(key):
  File "/usr/lib/python3.4/site-packages/salt/loader.py", line 1650, in _load
    ret = _inner_load(mod_name)
  File "/usr/lib/python3.4/site-packages/salt/loader.py", line 1639, in _inner_load
    if self._load_module(name) and key in self._dict:
  File "/usr/lib/python3.4/site-packages/salt/loader.py", line 1493, in _load_module
    mod.__opts__.update(self.opts)
  File "/usr/lib/python3.4/site-packages/salt/master.py", line 966, in _handle_signals
    channel.close()
  File "/usr/lib/python3.4/site-packages/salt/transport/zeromq.py", line 574, in close
    self.context.term()
  File "context.pyx", line 136, in zmq.backend.cython.context.Context.term (zmq/backend/cython/context.c:1866)
  File "checkrc.pxd", line 12, in zmq.backend.cython.checkrc._check_rc (zmq/backend/cython/context.c:2607)
  File "/usr/lib/python3.4/site-packages/salt/master.py", line 967, in _handle_signals
    super(MWorker, self)._handle_signals(signum, sigframe)
  File "/usr/lib/python3.4/site-packages/salt/utils/process.py", line 797, in _handle_signals
    sys.exit(salt.defaults.exitcodes.EX_OK)
SystemExit: 0
```
Minion Trace Log:
```
2018-06-21 11:58:01,087 [salt.utils.schedule:35  ][TRACE   ][11756] ==== evaluating schedule now None =====
2018-06-21 11:58:02,087 [salt.utils.schedule:35  ][TRACE   ][11756] ==== evaluating schedule now None =====
2018-06-21 11:58:02,358 [salt.transport.zeromq:1084][DEBUG   ][7080] SaltReqTimeoutError, retrying. (7/7)
2018-06-21 11:58:02,918 [salt.transport.zeromq:1084][DEBUG   ][7292] SaltReqTimeoutError, retrying. (6/7)
2018-06-21 11:58:03,087 [salt.utils.schedule:35  ][TRACE   ][11756] ==== evaluating schedule now None =====
2018-06-21 11:58:04,088 [salt.utils.schedule:35  ][TRACE   ][11756] ==== evaluating schedule now None =====
2018-06-21 11:58:04,746 [salt.transport.zeromq:1054][DEBUG   ][1076] Re-init ZMQ socket: Message timed out
2018-06-21 11:58:04,768 [salt.transport.zeromq:1029][DEBUG   ][1076] Trying to connect to: tcp://MASTERIP:4506
2018-06-21 11:58:04,771 [salt.utils.process:754 ][ERROR   ][1076] An un-handled exception from the multiprocessing process 'SignalHandlingMultiprocessingProcess-1' was caught:
Traceback (most recent call last):
  File "c:\salt\bin\lib\site-packages\salt\crypt.py", line 690, in sign_in
    timeout=timeout
  File "c:\salt\bin\lib\site-packages\tornado\gen.py", line 1055, in run
    value = future.result()
  File "c:\salt\bin\lib\site-packages\tornado\concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "<string>", line 4, in raise_exc_info
  File "c:\salt\bin\lib\site-packages\tornado\gen.py", line 1063, in run
    yielded = self.gen.throw(*exc_info)
  File "c:\salt\bin\lib\site-packages\salt\transport\zeromq.py", line 323, in send
    ret = yield self._uncrypted_transfer(load, tries=tries, timeout=timeout)
  File "c:\salt\bin\lib\site-packages\tornado\gen.py", line 1055, in run
    value = future.result()
  File "c:\salt\bin\lib\site-packages\tornado\concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "<string>", line 4, in raise_exc_info
  File "c:\salt\bin\lib\site-packages\tornado\gen.py", line 1063, in run
    yielded = self.gen.throw(*exc_info)
  File "c:\salt\bin\lib\site-packages\salt\transport\zeromq.py", line 312, in _uncrypted_transfer
    tries=tries,
  File "c:\salt\bin\lib\site-packages\tornado\gen.py", line 1055, in run
    value = future.result()
  File "c:\salt\bin\lib\site-packages\tornado\concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "<string>", line 4, in raise_exc_info
salt.exceptions.SaltReqTimeoutError: Message timed out

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\salt\bin\lib\site-packages\salt\utils\process.py", line 747, in _run
    return self._original_run()
  File "c:\salt\bin\lib\multiprocessing\process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "c:\salt\bin\lib\site-packages\salt\minion.py", line 1540, in _target
    Minion._thread_return(minion_instance, opts, data)
  File "c:\salt\bin\lib\site-packages\salt\minion.py", line 1697, in _thread_return
    timeout=minion_instance._return_retry_timer()
  File "c:\salt\bin\lib\site-packages\salt\minion.py", line 1920, in _return_pub
    ret_val = self._send_req_sync(load, timeout=timeout)
  File "c:\salt\bin\lib\site-packages\salt\minion.py", line 1362, in _send_req_sync
    return channel.send(load, timeout=timeout)
  File "c:\salt\bin\lib\site-packages\salt\utils\async.py", line 64, in wrap
    ret = self._block_future(ret)
  File "c:\salt\bin\lib\site-packages\salt\utils\async.py", line 74, in _block_future
    return future.result()
  File "c:\salt\bin\lib\site-packages\tornado\concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "<string>", line 4, in raise_exc_info
  File "c:\salt\bin\lib\site-packages\tornado\gen.py", line 1063, in run
    yielded = self.gen.throw(*exc_info)
  File "c:\salt\bin\lib\site-packages\salt\transport\zeromq.py", line 325, in send
    ret = yield self._crypted_transfer(load, tries=tries, timeout=timeout, raw=raw)
  File "c:\salt\bin\lib\site-packages\tornado\gen.py", line 1055, in run
    value = future.result()
  File "c:\salt\bin\lib\site-packages\tornado\concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "<string>", line 4, in raise_exc_info
  File "c:\salt\bin\lib\site-packages\tornado\gen.py", line 1063, in run
    yielded = self.gen.throw(*exc_info)
  File "c:\salt\bin\lib\site-packages\salt\transport\zeromq.py", line 290, in _crypted_transfer
    yield self.auth.authenticate()
  File "c:\salt\bin\lib\site-packages\tornado\gen.py", line 1055, in run
    value = future.result()
  File "c:\salt\bin\lib\site-packages\tornado\concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "<string>", line 4, in raise_exc_info
  File "c:\salt\bin\lib\site-packages\salt\crypt.py", line 602, in _authenticate
    creds = yield self.sign_in(channel=channel)
  File "c:\salt\bin\lib\site-packages\tornado\gen.py", line 1055, in run
    value = future.result()
  File "c:\salt\bin\lib\site-packages\tornado\concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "<string>", line 4, in raise_exc_info
  File "c:\salt\bin\lib\site-packages\tornado\gen.py", line 1063, in run
    yielded = self.gen.throw(*exc_info)
  File "c:\salt\bin\lib\site-packages\salt\crypt.py", line 699, in sign_in
    raise SaltClientError('Attempt to authenticate with the salt master failed with timeout error')
```

### Versions Report
This occurs on both 2018.3.1 Py3 and 2017.7.2 Py3 minions. The Master is 2018.3.1. The Master has had multiple versions of Tornado attempted excluding 5.0 (due to Py3 issues). I've moved back to Tornado 4.2.1 for now as that's what my previous Master had on 2017.7.2.
```
Salt Version:
           Salt: 2018.3.1

Dependency Versions:
           cffi: 1.11.5
       cherrypy: unknown
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.5.1
   mysql-python: Not Installed
      pycparser: 2.18
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.4.8 (default, Mar 23 2018, 10:04:27)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.4.3
            ZMQ: 4.1.4

System Versions:
           dist: centos 7.5.1804 Core
         locale: UTF-8
        machine: x86_64
        release: 3.10.0-862.3.3.el7.x86_64
         system: Linux
        version: CentOS Linux 7.5.1804 Core

```

**Comment by TheVakman** [comment]
I've tried to block communication via firewall so minions are unable to connect to fix them all breaking and allowed individual ones to test. The individual ones work A-OK. As soon as I allow the remaining minions, even the working ones start spawning python.exe processes.

**Comment by TheVakman** [comment]
Alternatively, it's related to this minion traceback that also occurs around the same time. Again, even on one that was working when other minions were blocked from connecting.

```
2018-06-21 09:43:54,572 [salt.log.setup   ][ERROR   ][12572] An un-handled exception was caught by salt's global exception handler:
AttributeError: cffi library '_openssl' has no function, constant or global variable named 'Cryptography_HAS_NETBSD_D1_METH'
Traceback (most recent call last):
  File "Scripts\salt-minion", line 26, in <module>
    salt_minion()
  File "C:\salt\bin\lib\site-packages\salt\scripts.py", line 168, in salt_minion
    minion.start()
  File "C:\salt\bin\lib\site-packages\salt\cli\daemons.py", line 340, in start
    super(Minion, self).start()
  File "C:\salt\bin\lib\site-packages\salt\utils\parsers.py", line 1032, in start
    self.prepare()
  File "C:\salt\bin\lib\site-packages\salt\cli\daemons.py", line 306, in prepare
    import salt.minion
  File "C:\salt\bin\lib\site-packages\salt\minion.py", line 90, in <module>
    import salt.pillar
  File "C:\salt\bin\lib\site-packages\salt\pillar\__init__.py", line 18, in <module>
    import salt.fileclient
  File "C:\salt\bin\lib\site-packages\salt\fileclient.py", line 29, in <module>
    import salt.utils.templates
  File "C:\salt\bin\lib\site-packages\salt\utils\templates.py", line 31, in <module>
    import salt.utils.http
  File "C:\salt\bin\lib\site-packages\salt\utils\http.py", line 71, in <module>
    import requests
  File "C:\salt\bin\lib\site-packages\requests\__init__.py", line 52, in <module>
    from .packages.urllib3.contrib import pyopenssl
  File "C:\salt\bin\lib\site-packages\requests\packages\urllib3\contrib\pyopenssl.py", line 46, in <module>
    import OpenSSL.SSL
  File "C:\salt\bin\lib\site-packages\OpenSSL\__init__.py", line 8, in <module>
    from OpenSSL import rand, crypto, SSL
  File "C:\salt\bin\lib\site-packages\OpenSSL\rand.py", line 12, in <module>
    from OpenSSL._util import (
  File "C:\salt\bin\lib\site-packages\OpenSSL\_util.py", line 6, in <module>
    from cryptography.hazmat.bindings.openssl.binding import Binding
  File "C:\salt\bin\lib\site-packages\cryptography\hazmat\bindings\openssl\binding.py", line 141, in <module>
    Binding.init_static_locks()
  File "C:\salt\bin\lib\site-packages\cryptography\hazmat\bindings\openssl\binding.py", line 122, in init_static_locks
    cls._ensure_ffi_initialized()
  File "C:\salt\bin\lib\site-packages\cryptography\hazmat\bindings\openssl\binding.py", line 109, in _ensure_ffi_initialized
    cls.lib = build_conditional_library(lib, CONDITIONAL_NAMES)
  File "C:\salt\bin\lib\site-packages\cryptography\hazmat\bindings\openssl\binding.py", line 69, in build_conditional_library
    if not getattr(lib, condition):
AttributeError: cffi library '_openssl' has no function, constant or global variable named 'Cryptography_HAS_NETBSD_D1_METH'
```

**Comment by TheVakman** [comment]
I believe a "pip3 install cryptography" has fixed this. I'm still not clear on why when blocking others from connecting, the couple ones I was allowing to connect didn't have an issue and why the couple allowed broke after allowing the remainder.

**Comment by TheVakman** [comment]
The latest update in the OP contains a trace log for the minion. I believe these are the ones causing them all to have issues.

**Comment by garethgreenaway** [comment]
@dwoz Thoughts on this one?

**Comment by TheVakman** [comment]
If it helps, I think the only relevant one is the minion trace in the OP now, I think "pip install cryptography" fixed a couple of the other items.

It's all very strange how allowing them to connect can cause problems for other ones that were working when the others were blocked from connecting.

Additionally, I notice sometimes I can stop the master for a bit, turn it back on and the same items will get logged but the other minions won't have an issue.

**Comment by TheVakman** [comment]
I don't see any discernible pattern to when this occurs but it's a huge problem when it does as it basically locks up the machine at 100% CPU.

**Comment by TheVakman** [comment]
Minion Version Report:
```
   Salt Version:
               Salt: 2018.3.1

    Dependency Versions:
               cffi: 1.10.0
           cherrypy: 10.2.1
           dateutil: 2.6.1
          docker-py: Not Installed
              gitdb: 2.0.3
          gitpython: 2.1.3
              ioflo: Not Installed
             Jinja2: 2.9.6
            libgit2: Not Installed
            libnacl: Not Installed
           M2Crypto: Not Installed
               Mako: 1.0.6
       msgpack-pure: Not Installed
     msgpack-python: 0.4.8
       mysql-python: Not Installed
          pycparser: 2.17
           pycrypto: 2.6.1
       pycryptodome: Not Installed
             pygit2: Not Installed
             Python: 3.5.3 (v3.5.3:1880cb95a742, Jan 16 2017, 16:02:32) [MSC v.1900 64 bit (AMD64)]
       python-gnupg: 0.4.1
             PyYAML: 3.12
              PyZMQ: 16.0.3
               RAET: Not Installed
              smmap: 2.0.3
            timelib: 0.2.4
            Tornado: 4.5.1
                ZMQ: 4.1.6

    System Versions:
               dist:
             locale: cp1252
            machine: AMD64
            release: 10
             system: Windows
            version: 10 10.0.14393 SP0 Multiprocessor Free
```

**Comment by TheVakman** [comment]
@Ch3LL Unfortunately, I don't think the referenced issue is related.

I'm pretty much at a loss with this one. I've got my master firewalled off to one minion right now so the minions can be used.

**Comment by TheVakman** [comment]
I've tried upgrading to 2018.3.2 on a minion and on the master side now and this still persists.

**Comment by TheVakman** [comment]
![image](https://user-images.githubusercontent.com/18545156/42286937-7adbe7ee-7f82-11e8-961a-7de2b56f71ca.png)

**Comment by TheVakman** [comment]
I've switched Tornado to 4.2.1, Jinja2 to 2.7.2, msgpack-python to 0.4.6 and restarted the Master. I'll see how it is in the morning as most users have gone offline for the day. CPU is OK for the ones currently online.

**Comment by rlwchang** [comment]
Any updates on this one? I'm also having a similar issue to this. It's also hard for me to reproduce, but it is pretty bad when it happens. Ignore the Node one as I know what's going on there.

My current setup is as follows:
Website -> Salt CherryPy Server -> Syndic -> 3 other Windows 10 machines with Salt

At startup, I often get the python2.7 from the master clocking in 100% CPU usage.
When running intensive tasks, I also get the Python 3 or syndic one clocking in 100%.

The screenshots below are from my syndic.  The syndic master is running inside a WSL using Ubuntu. The syndic minion is running in the native Windows 10 environment.

![image](https://user-images.githubusercontent.com/22105121/44555614-f226fe80-a6ea-11e8-90ca-a15a021da5ee.png)


![image](https://user-images.githubusercontent.com/22105121/44552880-8c357980-a6e0-11e8-9c9d-af70e372242a.png)


This is my versions report for the salt-master:
```
Salt Version:
           Salt: 2018.3.2

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.4.2
      docker-py: Not Installed
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (default, Dec  4 2017, 14:50:18)
   python-gnupg: 0.3.8
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: Ubuntu 16.04 xenial
         locale: UTF-8
        machine: x86_64
        release: 4.4.0-17134-Microsoft
         system: Linux
        version: Ubuntu 16.04 xenial
```

This is the versions report for the syndic
```
Salt Version:
           Salt: 2018.3.2

Dependency Versions:
           cffi: 1.10.0
       cherrypy: 10.2.1
       dateutil: 2.6.1
      docker-py: Not Installed
          gitdb: 2.0.3
      gitpython: 2.1.3
          ioflo: Not Installed
         Jinja2: 2.9.6
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.6
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: 2.17
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.5.3 (v3.5.3:1880cb95a742, Jan 16 2017, 16:02:32) [MSC v.1900 64 bit (AMD64)]
   python-gnupg: 0.4.1
         PyYAML: 3.12
          PyZMQ: 16.0.3
           RAET: Not Installed
          smmap: 2.0.3
        timelib: 0.2.4
        Tornado: 4.5.1
            ZMQ: 4.1.6

System Versions:
           dist:
         locale: cp1252
        machine: AMD64
        release: 10
         system: Windows
        version: 10 10.0.17134 SP0 Multiprocessor Free
```

**Comment by JonathanWilbur** [comment]
TLDR: Upgrade PyZMQ, check minion configuration, and use batching.

I had this same problem. I noticed that the salt-master processes would stay hot for a few minutes after running a big salt job (running `saltutil.sync_grains` on about 380 hosts, to be more precise), which made me think that there was probably a lot of I/O buffered that was being handled after the job. I imagine you may have noticed, as I have, that `/var/log/salt/master` is getting slammed by those error messages you reported. On my machine, the Salt Master dies if I do almost anything to over 100 minions at a time; I get that "The salt-master stopped responding. You may need to run this command with the --async flag..." warning when this happens.

Most of the files in the stacktrace were in the PyZMQ module, so I started there with upgrades. I upgraded PyZMQ out of band using `pip`, which put me on version 17.1.2. It seems to have made both the swarm of errors in the logs go away as well as those "--async" failures. Strangely, I don't have _any_ log messages now, but I guess that's better than the salt master dying.

Now a ton of minions do not return, even though the salt minion is running on them just fine. I believe this is still because the salt master is congested. It is basically DDoS'ed by all the minions responding all at once.

I would read [this documentation](https://docs.saltstack.com/en/latest/topics/tutorials/intro_scale.html), which may help with your performance issues. After that, I would just start using the `--batch-size` flag from the `salt` commands.

**Comment by alexz-kh** [comment]
Same issue here, ~50 minions. salt-master at VM with 10VCPU 24Gb ram.
```
Salt Version:
           Salt: 2017.7.8
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: 3.5.0
       dateutil: 2.4.2
      docker-py: 1.9.0
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.21.1
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (default, Nov 12 2018, 14:36:49)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.1.4
 
System Versions:
           dist: Ubuntu 16.04 xenial
         locale: UTF-8
        machine: x86_64
        release: 4.15.0-43-generic
         system: Linux
        version: Ubuntu 16.04 xenial
```

**Comment by TheVakman** [comment]
@alexz-kh For me, to fix the issue I changed the versions of various components to match what was in the official Salt repository for my release via pip. You can view the original comment above: https://github.com/saltstack/salt/issues/48241#issuecomment-403249355

I suspect it was Tornado that solved it for me. Tornado to 4.2.1, Jinja2 to 2.7.2, msgpack-python to 0.4.6, cffi to 1.11.4.

I couldn't answer why but it's probably worth trying to see if it fixes it for you too. The other person upgraded PyZMQ.

**Comment by alexz-kh** [comment]
@TheVakman thx for help, but still w\o clear RC understanding - its finger in the sky ;)
(I had also try to play with tornado and 2017.7.7\8 versions  - no luck )
Lets hope for answer from maintainers ;(
